js/renovation/pager/pages-large.tsx

```diff
@@ -0,0 +1,143 @@
+import {
+  Component,
+  ComponentBindings,
+  JSXComponent,
+  Event,
+  OneWay,
+  Fragment,
+  InternalState,
+} from 'devextreme-generator/component_declaration/common';
+import Page, { PageProps } from './page';
+
+const PAGER_PAGE_SEPARATOR_CLASS = 'dx-separator';
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export const viewFunction = ({ pages }: LargePages) => {
+  const PagesMarkup = pages.map((pageProps) => {
+    if (pageProps !== null) {
+      const { index = 0, selected, onClick } = pageProps;
+      return (
+        <Page
+          key={index.toString()}
+          index={index}
+          selected={selected}
+          onClick={onClick}
+        />
+      );
+    }
+    return (
+      <div key="delimiter" className={PAGER_PAGE_SEPARATOR_CLASS}>. . .</div>
+    );
+  });
+  return (<Fragment>{PagesMarkup}</Fragment>);
+};
+
+@ComponentBindings()
+export class LargePagesProps {
+  @OneWay() maxPagesCount?: number = 10;
+
+  @OneWay() pageCount?: number = 10;
+
+  @OneWay() pageIndex?: number = 0;
+
+  @OneWay() rtlEnabled?: boolean = false;
+
+  @Event() pageIndexChange?: (pageIndex: number) => void = () => { }; // commonUtils.noop
```

Comment:
Why is this comment here?

Reply:
It's discussion able question. Should we initialize events? I add it in TODO list